common/web/gesture-recognizer/src/engine/headless/gestures/matchers/matcherSelector.ts

```diff
@@ -234,10 +234,23 @@ export class MatcherSelector<Type, StateToken = any> extends EventEmitter<EventM
     const unmatchedSource = sourceNotYetStaged ? source : null;
     const priorMatcher = sourceNotYetStaged ? null: source;
 
+    // matchGesture calls should be queued and act atomically, in sequence.
     if(this.pendingMatchSetup) {
+      const parentLockPromise = this.pendingMatchSetup;
```

Comment:
At one point in development, I added an extra component to this condition:  ` && sourceNotYetStaged`.

I expect that only new incoming sources would have their fate altered by ignoring this locking section; the initial state and item for continuing gestures won't be altered by ignoring it.  

That said... I cannot confidently say that an inconsistent locking scheme here would be without issues.  It would almost certainly be prone to more issues; it would likely have effects on multitaps, at a minimum.